sips/sip-x/sip-x-improved-stacking.md

```diff
@@ -0,0 +1,205 @@
+
+## Semantic Change of Delegation
+
+When signers verify and accept proposed blocks by miners, their voting power corresponds to the amount of stacked Stacks tokens (see SIP-021). The new stacking process changes the delegation flow as follows:
+
+- `delegate-stx` shall be renamed to `delegate`. The function takes the arguments: `amount`, `delegate-to`, `signature`, optional bitcoin `block-height` defining the end of stacking and auto extending, optional `max-amount`. The `amount` defines how many Stacks tokens are locked immediately. The `max-amount` the maximum of Stacks tokens that can be locked in the future through auto extending. If omitted the whole balance is locked. The argument `delegate-to` describes the signer. The argument `signature` is a signature of the signer indicating that the signer accepted the delegation of voting power.
+
+- `revoke-delegate-stx` shall be renamed to `revoke-delegate`. After calling this functions, auto extension is stopped and Stacks tokens are unlocked for the user at the end of the current cycle. If the signer the user delegated to did not aggregated enough Stacks tokens to receive at least one reward slot then the user's Stacks token are unlocked immediately through this call.
+
+- `delegate-increase` is replaced by `delegate`. Users can simply call delegate with a higher amount.
+
+- `delegate-extend` is removed because the locking period is automatically extended each cycle.
+
+## Aggregation Commit and PoX Reward Address
+
+Signers provide a PoX reward address during the aggregation commit call. This bitcoin address shall be used to receive stacking rewards.
+
+There is no change in the registration of PoX reward addresses. In particular, there can be more than one PoX reward address per signer.
+
+Stacks tokens that are part of a PoX reward address that aggregated less than the minimum required for at least one reward slot can be unlocked through `revoke-delegate`.
+
+## Relation between Stackers and Signers
+
+The signatures provided by signers to stackers shall use one of the following topics
+
+- delegate
+- agg-commit
+- agg-increase
+
+The other topics are no longer used.
+
+## Registration of Signers
+
+Signers handle three private keys:
+
+1. one for signing blocks and delegation approval;
+2. one for PoX reward address and reward distribution;
+3. one for aggregation on the Stacks blockchain.
+
+The Stacks address of the first key is used by Stacks holders during delegation for `delegate-to`.
+
+The encoded bitcoin address of the second key and the Stacks address of the third key are announced in a registration contract call. The new function of the PoX contract is named `signer-register` and just takes the encoded bitcoin address, and the stacks address as parameters.
+
+Note, for solo stackers these three keys can be just a single key. For more complex setup, the keys can be handled by different independant entities. Also, the PoX reward address can be a deposit address for sBTC.
```

Comment:
...independent...

(instead of _independ**a**nt_)

Reply:
:+1: